sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -166,6 +197,25 @@ class FileStreamSource(
         // implies "sourceOptions.latestFirst = true" which we want to refresh the list per batch
         (newFiles.take(files.maxFiles()), null)
 
+      case files: ReadMaxBytes if !sourceOptions.latestFirst =>
+        // we can cache and reuse remaining fetched list of files in further batches
+        val (FilesSplit(bFiles, _), FilesSplit(usFiles, rSize)) =
+          takeFilesUntilMax(newFiles, files.maxBytes())
+        if (rSize < (files.maxBytes() * DISCARD_UNSEEN_FILES_RATIO).toLong) {
```

Comment:
`.toLong` is required here?

Reply:
Yes, well cast need to be here somewhere otherwise a compiler throws: 
```
type mismatch;
 found   : Long
 required: ?{def *(x$1: ? >: Double): ?}
Note that implicit conversions are not applicable because they are ambiguous:
 both method long2float in object Long of type (x: Long): Float
 and method long2double in object Long of type (x: Long): Double
 are possible conversion functions from Long to ?{def *(x$1: ? >: Double): ?}
        if (rSize < files.maxBytes() * DISCARD_UNSEEN_FILES_RATIO) {
```